File: yep/src/main/java/catchla/yep/view/holder/DribbbleTopicViewHolder.java
package catchla.yep.view.holder;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import catchla.yep.R;
import catchla.yep.adapter.TopicsAdapter;
import catchla.yep.model.DribbbleAttachment;
import catchla.yep.model.Topic;
import catchla.yep.util.ImageLoaderWrapper;

/**
 * Created by mariotaku on 15/12/9.
 */
public class DribbbleTopicViewHolder extends TopicViewHolder {

    private final ImageView mediaPreviewView;
    private final TextView titleView;

    public DribbbleTopicViewHolder(final TopicsAdapter topicsAdapter, final View itemView, final Context context,
                                   final ImageLoaderWrapper imageLoader,
                                   final TopicsAdapter.TopicClickListener listener) {
        super(topicsAdapter, itemView, context, imageLoader, listener);
        itemView.findViewById(R.id.attachment_view).setOnClickListener(this);
        mediaPreviewView = (ImageView) itemView.findViewById(R.id.media_preview);
        titleView = (TextView) itemView.findViewById(R.id.title);
    }

    @Override
    public void displayTopic(final Topic topic) {
        super.displayTopic(topic);
        DribbbleAttachment attachment = getDribbbleAttachment(topic);
        getImageLoader().displayImage(attachment.getMediaUrl(), mediaPreviewView);
        titleView.setText(attachment.getTitle());
    }

    private DribbbleAttachment getDribbbleAttachment(final Topic topic) {
        return (DribbbleAttachment) topic.getAttachments().get(0);
    }

    @Override
    protected void onAttachmentClick() {
        final DribbbleAttachment attachment = getDribbbleAttachment(adapter.getTopic(getLayoutPosition()));
        adapter.getContext().startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(attachment.getUrl())));
    }
}

File: src/main/kotlin/catchla/yep/view/holder/DribbbleTopicViewHolder.kt
package catchla.yep.view.holder

import android.app.Activity
import android.content.Context
import android.net.Uri
import android.view.View
import catchla.yep.adapter.TopicsAdapter
import catchla.yep.model.DribbbleAttachment
import catchla.yep.model.Topic
import catchla.yep.util.ImageLoaderWrapper
import catchla.yep.util.Utils
import catchla.yep.util.view.ViewSupport
import kotlinx.android.synthetic.main.layout_topic_attachment_dribbble.view.*

/**
 * Created by mariotaku on 15/12/9.
 */
class DribbbleTopicViewHolder(topicsAdapter: TopicsAdapter, itemView: View, context: Context,
                              imageLoader: ImageLoaderWrapper,
                              listener: TopicsAdapter.TopicClickListener?) : TopicViewHolder(topicsAdapter, itemView, context, imageLoader, listener) {
    private val attachmentView by lazy { itemView.attachmentView }
    private val mediaPreviewView by lazy { itemView.mediaPreview }
    private val titleView by lazy { itemView.title }

    init {
        attachmentView.setOnClickListener(this)
        ViewSupport.setClipToOutline(attachmentView, true)
    }

    override fun displayTopic(topic: Topic) {
        super.displayTopic(topic)
        val attachment = getDribbbleAttachment(topic)
        imageLoader.displayImage(attachment.mediaUrl, mediaPreviewView)
        titleView.text = attachment.title
    }

    private fun getDribbbleAttachment(topic: Topic): DribbbleAttachment {
        return topic.attachments[0] as DribbbleAttachment
    }

    override fun onAttachmentClick() {
        val attachment = getDribbbleAttachment(adapter.getTopic(layoutPosition))
        Utils.openUri(adapter.context as Activity, Uri.parse(attachment.url))
    }
}
